### PR TITLE
Remove unused request parameter from controller functions

### PR DESCRIPTION
**Pull request by michaelestrin**
Removed unused request parameters from controller functions.  Cleaned
up formatting. Cleaned up io.WriteString() usage; now consistently
using w.Write(). Explicitly ignoring errors from w.Write() and
r.Body.Close().  Folded metrics, configuration, and ping controller
functions into routing definition closures (except for export
services which are being archived before Geneva release).

Fixes: https://github.com/edgexfoundry/edgex-go/issues/2074

Signed-off-by: Michael Estrin <m.estrin@dell.com>

### CODE REVIEW

**Comment by michaelestrin** [comment]
Rebased and addressed conflicts.

**Comment by michaelestrin** [comment]
Rebased.

File: internal/core/data/router.go [hunk 1]
```diff
@@ -774,7 +821,7 @@ func readingHandler(
 
 		pkg.Encode(r, w, loggingClient)
 	case http.MethodPost:
-		reading, err := decodeReading(r.Body, loggingClient, dbClient)
+		r, err := decodeReading(r.Body, loggingClient, dbClient)
```
Comment (AnthonyMBonafide) [inline internal/core/data/router.go]: Can we undo the variable name change from `reading`?

File: internal/core/data/router.go [hunk 2]
```diff
@@ -857,14 +904,14 @@ func getReadingByIdHandler(
...
-	defer r.Body.Close()
+	defer func() { _ = r.Body.Close() }()
 
 	vars := mux.Vars(r)
 	id := vars["id"]
 
 	switch r.Method {
 	case http.MethodGet:
-		reading, err := getReadingById(id, loggingClient, dbClient)
+		r, err := getReadingById(id, loggingClient, dbClient)
```
Comment (AnthonyMBonafide) [inline internal/core/data/router.go]: See my other comment regarding reverting this name change.

File: internal/core/data/router.go (hunk 1 above)
Comment (michaelestrin) [reply internal/core/data/router.go]: Change was intentional; reading is a package name.  What I didn't realize was r was already in use.

**Comment by michaelestrin** [comment]
Added hold to flag merge must be squashed (can omit 2nd commit's message).

**Comment by michaelestrin** [comment]
Rebased, squashed, and resolved conflicts.

File: internal/core/data/router.go (hunk 1 above)
Comment (AnthonyMBonafide) [reply internal/core/data/router.go]: My apologies, I thought I put that into the original comment and instead left that unclear one.

**Comment by michaelestrin** [comment]
Rebased and addressed conflicts.

**Comment by michaelestrin** [comment]
Rebased and addressed conflicts.

**Comment by michaelestrin** [comment]
Rebased and addressed conflicts.

**Comment by michaelestrin** [comment]
Rebased and addressed conflicts.

**Pull request merged**